build_posix/aclocal/options.m4

```diff
@@ -215,6 +215,18 @@ pthread_adaptive|pthreads_adaptive)
 esac
 AC_MSG_RESULT($with_spinlock)
 
+AH_TEMPLATE(HAVE_STRICT, [Enable strict compiler checking.])
+AC_MSG_CHECKING(if --enable-strict option specified)
+AC_ARG_ENABLE(strict,
+	[AS_HELP_STRING([--enable-strict],
+	    [Enable strict compiler checking.])], r=$enableval, r=no)
+case "$r" in
+no)	wt_cv_enable_strict=no;;
+*)	AC_DEFINE(HAVE_STRICT)
```

Comment:
Can we do without a macro?  AFAICT, it isn't currently used and I'd be unhappy with having any code that we add just to fix the strict build.